nomt/src/beatree/ops/update/leaf_updater.rs

```diff
@@ -903,4 +909,265 @@ mod tests {
             ]
         );
     }
+
+    #[test]
+    fn split_left_node_below_target() {
+        let mut updater = LeafUpdater::new(PAGE_POOL.clone(), None, None);
+        let mut new_leaves = TestHandleNewLeaf::default();
+
+        updater.ingest(key(1), Some(vec![1; 1100]), false, |_| {});
+        updater.ingest(key(2), Some(vec![1; 1100]), false, |_| {});
+        updater.ingest(key(3), Some(vec![1; 1000]), false, |_| {});
+        updater.ingest(key(4), Some(vec![1; 1000]), false, |_| {});
+        updater.ingest(key(5), Some(vec![1; 1000]), false, |_| {});
+        updater.ingest(key(6), Some(vec![1; 1300]), false, |_| {});
+
+        let tot_body_size = updater.gauge.body_size();
+        let midpoint = tot_body_size / 2;
+
+        updater.try_build_leaves(&mut new_leaves, midpoint);
+
+        let leaf_1 = &new_leaves.inner.get(&[0; 32]).unwrap().0;
+        assert_eq!(leaf_1.n(), 3);
+        let leaf_body_size = body_size(leaf_1.n(), leaf_1.values_size(0, leaf_1.n()));
+        assert!(leaf_body_size < midpoint);
+        let leaf_2 = &new_leaves.inner.get(&separate(&key(3), &key(4))).unwrap().0;
+        assert_eq!(leaf_2.n(), 3);
+    }
+
+    #[test]
+    fn split_left_node_impossible_not_halfull() {
```

Comment:
half_full

(I've also learned the term "rightsized" to refer to nodes that are between the minimum and maximum)

Reply:
Thank you! `split_left_node_always_rightsized` is perfect!

Not sure why in this stack I was convinced that half-full is written as halfull and also other words around a similar error :sweat_smile: